src/components/Questions/QuestionModal.vue

```diff
@@ -115,243 +117,278 @@ export default defineComponent({
+      state.localCurrentQuestionIndex = props.questions.length
+      context.emit('end-test')
     }
 
     onUnmounted(() => {
       // remove listeners
-      window.removeEventListener("resize", checkScreenOrientation);
-    });
+      window.removeEventListener('resize', checkScreenOrientation)
+    })
 
     const currentQuestion = computed(
       () => props.questions[props.currentQuestionIndex]
-    );
+    )
 
-    const questionType = computed(() => currentQuestion.value.type);
+    const questionType = computed(() => currentQuestion.value.type)
 
     const questionCorrectAnswer = computed(
       () => currentQuestion.value?.correct_answer
-    );
+    )
 
-    const isGradedQuestion = computed(() => currentQuestion.value.graded);
+    const isGradedQuestion = computed(() => currentQuestion.value.graded)
 
     const isQuestionTypeMultiChoice = computed(
-      () => questionType.value == "multi-choice"
-    );
+      () => questionType.value == 'multi-choice'
+    )
     const isQuestionTypeSingleChoice = computed(
-      () => questionType.value == "single-choice"
-    );
+      () => questionType.value == 'single-choice'
+    )
     const isQuestionTypeSubjective = computed(
-      () => questionType.value == "subjective"
-    );
+      () => questionType.value == 'subjective'
+    )
+    const isQuestionTypeIntegerNumerical = computed(
+      () => questionType.value == 'numerical-integer'
+    )
+    const isQuestionTypeFloatNumerical = computed(
+      () => questionType.value == 'numerical-float'
+    )
 
     const currentQuestionResponse = computed(
       () => props.responses[props.currentQuestionIndex]
-    );
+    )
 
     const currentQuestionResponseAnswer = computed(
       () => currentQuestionResponse.value?.answer
-    );
+    )
 
     const isAnswerSubmitted = computed(() => {
-      if (currentQuestionResponseAnswer.value == null) return false;
+      if (currentQuestionResponseAnswer.value == null) return false
+      if (typeof currentQuestionResponseAnswer.value == 'number') {
+        return currentQuestionResponseAnswer.value != 0
+      }
       if (isQuestionTypeSingleChoice.value || isQuestionTypeMultiChoice.value) {
-        return currentQuestionResponseAnswer.value.length > 0;
+        return currentQuestionResponseAnswer.value.length > 0
       }
-      return true;
-    });
+      return true
+    })
 
     const isAttemptValid = computed(() => {
       const currentDraftResponse = state.draftResponses[
         props.currentQuestionIndex
-      ] as DraftResponse;
-      if (currentDraftResponse == null) return false;
-      if (isQuestionTypeSubjective.value) return currentDraftResponse != "";
-      return currentDraftResponse.length > 0;
-    });
+      ] as DraftResponse
+      if (currentDraftResponse == null) {
+        return false
+      }
+      if (isQuestionTypeSubjective.value) return currentDraftResponse != ''
+      if (typeof currentDraftResponse == 'number') {
```

Comment:
I have not done that because if I only check for ```isQuestionTypeIntegerNumerical || isQuestionTypeFloatNumerical``` the typescript shows an error that length can not be used for number type. So I have simply put a ```typeof```` check as only the numerical type questions will have number type.